homeassistant/components/weather/__init__.py

```diff
@@ -1088,6 +1121,56 @@ async def async_get_forecast_service(
     }
 
 
+async def async_get_forecast_attr_service(
+    weather: WeatherEntity, service_call: ServiceCall
+) -> ServiceResponse:
+    """Get weather forecast for specified attribute."""
+    forecast_attribute = service_call.data["attribute"]
+    forecast_limit_hours = service_call.data.get("limit_hours")
+
+    supported_features = weather.supported_features or 0
+    if (supported_features & WeatherEntityFeature.FORECAST_HOURLY) != 0:
+        native_forecast_list = await weather.async_forecast_hourly()
+    elif (supported_features & WeatherEntityFeature.FORECAST_TWICE_DAILY) != 0:
+        native_forecast_list = await weather.async_forecast_twice_daily()
+        if forecast_limit_hours is not None:
+            forecast_limit_hours = forecast_limit_hours // 12 + 1
+    elif (supported_features & WeatherEntityFeature.FORECAST_DAILY) != 0:
+        native_forecast_list = await weather.async_forecast_daily()
+        if forecast_limit_hours is not None:
+            forecast_limit_hours = forecast_limit_hours // 24 + 1
+    else:
+        raise HomeAssistantError(
```

Comment:
When can this happen? We require at least one of the supported features when registering the service.

Reply:
The function doesn't know this.

Reply:
If it's to satisfy typing we could raise eg `RuntimeError` instead and add a comment. `HomeAssistantError` is reserved for expected errors.

Reply:
Can remove the `else`-block?

Reply:
@MartinHjelmare thanks for explaining. I will switch to RuntimeError.
@gjohansson-ST No as it would be bad code.

Reply:
I meant more to rewrite it, maybe just "copy" how it's in `async_get_forecast_service()`.
Can break out that part into it's own function just to avoid duplicated code.